File: Coding Problems & My Solutions/Longest Substring Without Repeating Characters/Longest_Substring_Without_Repeating_Characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int max = 0;
        int i = 0;
        int j = 0;
        HashSet<Character> set = new HashSet<>();
        while (j < s.length() && i < s.length()){
            if (set.contains(s.charAt(j))){
                set.remove(s.charAt(i));
                i++;
            } else {
                set.add(s.charAt(j));
                j++;
                max = Math.max(max, j-i);
            }
        }
        return max;
    }
}

File: Coding Problems & My Solutions/Longest Substring Without Repeating Characters/Longest_Substring_Without_Repeating_Characters.swift
class Solution {
    func lengthOfLongestSubstring(_ s: String) -> Int {
        var maxLen = 0
        var set = Set<Character>()
        var arr = Array(s)
        var a = 0
        var b = 0
        while b < arr.count {
            if set.contains(arr[b]) {
                set.remove(arr[a])
                a += 1
            } else {
                set.insert(arr[b])
                maxLen = max(maxLen, b - a + 1)
                b += 1
            }
        }
        return maxLen
    }
}

class Solution {
    func lengthOfLongestSubstring(_ s: String) -> Int {
        var set = Set<Character>()
        var ans = 0
        let charArray = Array(s)//s.map { String($0) }
        var a = 0
        var b = 0
        while b < charArray.count {
            if set.contains(charArray[b]) {
                set.remove(charArray[a])
                a += 1
            } else {
                set.insert(charArray[b])
                let currentLen = (b - a) + 1
                ans = max(ans, currentLen)
                b += 1
            }
        }
        return ans
    }
}